BUILD.md

```diff
@@ -0,0 +1,66 @@
+# Build How-to
+
+This is a quick start guide for building nijigenerate and nijiexpose. For users and developers,
+you can follow the instructions below to build on your Linux system.
+
+Also See [build-and-release.yml in nijigenerate](https://github.com/nijigenerate/nijigenerate/blob/main/.github/workflows/build-and-release.yml) 
+and [build-and-release.yml in nijiexpose](https://github.com/nijigenerate/nijiexpose/blob/main/.github/workflows/build-and-release.yml)
+
+## Install Dependencies
+Ubuntu
+```bash
+sudo apt update
+sudo apt install ldc build-essential dub cmake git libsdl2-dev libfreetype6-dev
+```
+
+Archlinux
+```bash
+sudo pacman -Syu
+sudo pacman -S ldc base-devel dub cmake git sdl2 freetype2
+```
+
+## Build nijilive Project
+First, we need to clone the four projects under nijigenerate and add them to `dub add-local`.
+```bash
+git clone https://github.com/nijigenerate/nijigenerate
+dub add-local ./nijigenerate 0.0.1
+git clone https://github.com/nijigenerate/nijilive
+dub add-local ./nijilive 0.0.1
+git clone https://github.com/nijigenerate/nijiui
+dub add-local ./nijiui 0.0.1 
+git clone https://github.com/nijigenerate/nijiexpose
+dub add-local ./nijiexpose 0.0.1
+```
+
+dub package `i2d-imgui` will use the `Master` branch by default when building SDL2,
+but the upstream branch has been changed to `main`. We need to fix it with the following instructions.
+```bash
+git clone https://github.com/Inochi2D/i2d-imgui
+dub add-local i2d-imgui 0.8.0
+```
+
+On Debian/Ubuntu, a link error may occur. To fix the libz issue, use `vim ./nijigenerate/dub.sdl` to find this line:
```

Comment:
I didn't experience this in my environment. Could you tell me your build environment?
I use dmd instead of ldc in my environment. I think CI/CD uses ldc2.

Reply:
I rechecked, and the problem occurs in Ubuntu 24.04 LTS. Use the following instructions to create the Ubuntu container. This is equivalent to a clean Ubuntu 24.04 installation.
```bash
docker run --platform linux/amd64 -it ubuntu /bin/bash
```
If you don't edit `dub.sdl`, there will be a link error. Some people have encountered the same issue on Discord. You should be able to search for `lz` in the #general channel to find it.
```
 Linking nijigenerate
/usr/bin/ld: ../.dub/cache/imagefmt/2.1.2/build/library-release-Kz80jNsq1VPNxGoGjh2aDA/libimagefmt.a(imagefmt.png.o): undefined reference to symbol 'inflateEnd'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/13/../../../x86_64-linux-gnu/libz.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
Error: /usr/bin/cc failed with status: 1
Error ldc2 failed with exit code 1.
```